### PR TITLE
[mlrun] add max-connections to mysql db

### PR DESCRIPTION
**Pull request by Tankilevitch**
### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove irrelevant fields.]
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### Description:
[Please add a description of the change request.]

### CODE REVIEW

File: stable/mlrun/templates/db-configmap.yaml [hunk 1]
```diff
@@ -49,14 +49,15 @@ data:
...
+      --max-connections={{- .Values.db.dbConfiguration.maxConnections }} \
+      --innodb-adaptive-hash-index={{- .Values.db.dbConfiguration.adaptiveHashIndex }} \
+      --innodb-read-io-threads={{- .Values.db.dbConfiguration.readIOThreads }} \
+      --innodb-write-io-threads={{- .Values.db.dbConfiguration.writeIOThreads }} \
+      --innodb-purge-threads={{- .Values.db.dbConfiguration.purgeThreads }} \
+      --innodb-buffer-pool-size={{- .Values.db.dbConfiguration.bufferPoolSize }} \
+      --innodb-buffer-pool-instances={{- .Values.db.dbConfiguration.bufferPoolInstances }} \
+      --innodb-page-cleaners={{- .Values.db.dbConfiguration.pageCleaners }} \
+      --innodb-io-capacity={{- .Values.db.dbConfiguration.IOCapacity }} \
       --innodb-flush-neighbors=0 \
```
Comment (alonmr) [inline stable/mlrun/templates/db-configmap.yaml]: looks like you missed one
```suggestion
      --innodb-flush-neighbors={{- .Values.db.dbConfiguration.flushNeighbors }} \
```

**Review by alonmr — COMMENTED** [review]
minor comment

**Comment by alonmr** [comment]
You might want to use `_helpers.tpl` to not break backwards compatibility
For example:
```
{{- define "mlrun.db.dbConfiguration.adaptiveHashIndex" -}}
{{ default .Values.db.dbConfiguration.adaptiveHashIndex 0 }}
{{- end -}}
```

And then you would use:
```
--innodb-adaptive-hash-index={{ template "mlrun.db.dbConfiguration.adaptiveHashIndex" . }} \
```
Otherwise you would now need to add these values to provazio and such since it overrides the values file.

File: stable/mlrun/values.yaml [hunk 2]
```diff
@@ -385,6 +385,27 @@ db:
...
+    # mysql default is 151, we raised the connection number to handle the scale of MLRun multi instance architecture
+    # ideally should be calculated by : <number-of-mlrun-instances> * <max-workers> + 1 ( for SUPER account )
+    # SUPER is a MySQL privilege that grants admin rights to the user.
+    maxConnections: 512
+
+    # refer to https://github.com/v3io/helm-charts/pull/674
+    adaptiveHashIndex: 0
+    readIOThreads: 32
+    writeIOThreads: 32
+    purgeThreads: 16
```
Comment (liranbg) [inline stable/mlrun/values.yaml]: nest under "innoDB"

File: stable/mlrun/values.yaml [hunk 3]
```diff
@@ -385,6 +385,27 @@ db:
...
+
+    # refer to https://github.com/v3io/helm-charts/pull/671
+    bufferPoolSize: 1073741824
+    bufferPoolInstances: 8
```
Comment (liranbg) [inline stable/mlrun/values.yaml]: nest under "innoDB"

File: stable/mlrun/values.yaml [hunk 4]
```diff
@@ -385,6 +385,27 @@ db:
...
+
+    # refer to https://github.com/v3io/helm-charts/pull/680
+    pageCleaners: 8
+    IOCapacity: 10000
+    flushNeighbors: 0
```
Comment (liranbg) [inline stable/mlrun/values.yaml]: nest under "innoDB"

**Pull request merged**